test/files/pos/S1.scala

```diff
@@ -8,24 +8,6 @@
 **           ^
 */
 class S1() {
-  def foo[T <: this.type](x: T) = x
-  def f = foo[this.type](this)
+    def foo[T <: this.type](x: T) = x;
+    foo[this.type](this);
 }
-
-class S2() {
-  def foo[T <: this.type](x: T) = x
-  def f = foo(this)
-}
```

Comment:
Why was this removed? It still compiles after this PR.

It would also be a good opportunity to rename the file of this test :-) And the comment in the beginning looks stale.

Reply:
Hmm, the test failed on my machine, but maybe that was spurious.

Reply:
I tried with `scala-pr 6789` in the repl.

Reply:
Maybe I had a stray class file or something. Pushed a revert for it